File: codeforces/1903/C.py
def calculate_maximum_sum():
    # Input: Read array size
    array_size = int(input())
    
    # Input: Read array elements
    array = list(map(int, input().split()))

    # Initialize variables
    remaining_sum = 0
    dynamic_programming = [(0, 1) for _ in range(array_size + 1)]
    dynamic_programming[array_size] = (0, 1)

    # Dynamic Programming loop to calculate maximum sum
    for i in range(array_size - 1, -1, -1):
        total, count = dynamic_programming[i + 1]
        dynamic_programming[i] = (total + array[i], count)
        dynamic_programming[i] = max(dynamic_programming[i], (total + remaining_sum + array[i], count + 1))
        remaining_sum += array[i]

    # Return the maximum sum for the current test case
    return dynamic_programming[0][0]


def main():
    # Input: Read the number of test cases
    test_cases = int(input())

    # Process each test case
    for _ in range(test_cases):
        result = calculate_maximum_sum()

        # Output: Print the result for each test case
        print(result)


if __name__ == "__main__":
    main()

File: codeforces/1903/C.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

int calculateMaximumSum()
{
    // Input: Read array size
    int arraySize;
    cin >> arraySize;

    // Input: Read array elements
    vector<int> array(arraySize);
    for (int i = 0; i < arraySize; ++i)
    {
        cin >> array[i];
    }

    // Initialize variables
    int remainingSum = 0;
    vector<pair<int, int>> dynamicProgramming(arraySize + 1, make_pair(0, 1));
    dynamicProgramming[arraySize] = make_pair(0, 1);

    // Dynamic Programming loop to calculate maximum sum
    for (int i = arraySize - 1; i >= 0; --i)
    {
        int total = dynamicProgramming[i + 1].first;
        int count = dynamicProgramming[i + 1].second;
        dynamicProgramming[i] = make_pair(total + array[i], count);
        dynamicProgramming[i] = max(dynamicProgramming[i], make_pair(total + remainingSum + array[i], count + 1));
        remainingSum += array[i];
    }

    // Return the maximum sum for the current test case
    return dynamicProgramming[0].first;
}

int32_t main()
{
    // Input: Read the number of test cases
    int testCases;
    cin >> testCases;

    // Process each test case
    for (int i = 0; i < testCases; ++i)
    {
        int result = calculateMaximumSum();

        // Output: Print the result for each test case
        cout << result << endl;
    }

    return 0;
}
